modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -204,7 +204,13 @@ def __init__(
         self._dtypes = dtypes
 
         self._validate_axes_lengths()
-        self._filter_empties(compute_metadata=False)
+        if all(obj is not None for obj in (index, columns, row_lengths, column_widths)):
+            # this hint allows to filter empty partitions out without triggering metadata computation
+            # in order to avoid useless computation over empty partitions
+            compute_metadata = True
+        else:
+            compute_metadata = False
+        self._filter_empties(compute_metadata=compute_metadata)
```

Comment:
```suggestion
        if all(obj is not None for obj in (index, columns, row_lengths, column_widths)):
            # this hint allows to filter empty partitions out without triggering metadata computation
            # in order to avoid useless computation over empty partitions
            compute_metadata = True
        else:
            compute_metadata = False
        self._filter_empties(compute_metadata=False)
```